### PR TITLE
Get rid of some duplicate code.

### PR DESCRIPTION
**Pull request by shlomif**
Refactoring. I hereby put my changes under
https://creativecommons.org/choose/zero/ but giving me credit (see
http://www.shlomifish.org/meta/copyrights/ ) would be appreciated.

### CODE REVIEW

File: hero.js [hunk 1]
```diff
@@ -9,20 +9,17 @@ var Hero = function(x, y, variant, facing, map) {
...
-  this.animations.add(this.getAnimName(DIRECTION.LEFT, colorVariant.GREEN), [27, 28, 29], 12, true);
-  this.animations.add(this.getAnimName(DIRECTION.RIGHT, colorVariant.GREEN), [30, 31, 32], 12, true);
-  this.animations.add(this.getAnimName(DIRECTION.UP, colorVariant.GREEN), [33, 34, 35], 12, true);
+  var t = this;
+  var i = -1;
+  [colorVariant.RED, colorVariant.BLUE, colorVariant.GREEN].forEach(function (color) {
+    [DIRECTION.DOWN, DIRECTION.LEFT, DIRECTION.RIGHT, DIRECTION.UP].forEach(function (dir) {
+      var arr = [];
+      for (var j=0; j < 3 ; ++j) {
+        arr.push((++i) + 0);
```
Comment (satanas) [inline hero.js]: Why the `+ 0`? I think we don't need coercion here

**Comment by shlomif** [comment]
On Sat, 19 Aug 2017 09:53:54 -0700
Wil Alvarez <notifications@github.com> wrote:

> satanas commented on this pull request.
> 
> 
> 
> > -  this.animations.add(this.getAnimName(DIRECTION.DOWN, colorVariant.BLUE),
> > [12, 13, 14], 12, true);  
> -  this.animations.add(this.getAnimName(DIRECTION.LEFT, colorVariant.BLUE),
> [15, 16, 17], 12, true);
> -  this.animations.add(this.getAnimName(DIRECTION.RIGHT, colorVariant.BLUE),
> [18, 19, 20], 12, true);
> -  this.animations.add(this.getAnimName(DIRECTION.UP, colorVariant.BLUE),
> [21, 22, 23], 12, true); -
> -  this.animations.add(this.getAnimName(DIRECTION.DOWN, colorVariant.GREEN),
> [24, 25, 26], 12, true);
> -  this.animations.add(this.getAnimName(DIRECTION.LEFT, colorVariant.GREEN),
> [27, 28, 29], 12, true);
> -  this.animations.add(this.getAnimName(DIRECTION.RIGHT, colorVariant.GREEN),
> [30, 31, 32], 12, true);
> -  this.animations.add(this.getAnimName(DIRECTION.UP, colorVariant.GREEN),
> [33, 34, 35], 12, true);
> +  var t = this;
> +  var i = -1;
> +  [colorVariant.RED, colorVariant.BLUE, colorVariant.GREEN].forEach(function
> (color) {
> +    [DIRECTION.DOWN, DIRECTION.LEFT, DIRECTION.RIGHT,
> DIRECTION.UP].forEach(function (dir) {
> +      var arr = [];
> +      for (var j=0; j < 3 ; ++j) {
> +        arr.push((++i) + 0);
> 
> Why the `+ 0`? I think we don't need coercion here
> 

I wanted to be extra sure we get a copy of the value instead of reusing the
same reference (and value). But it may not be needed.

-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/
Why I Love Perl - http://shlom.in/joy-of-perl

Chuck Norris can make East and West meet.
    — http://www.shlomifish.org/humour/bits/facts/Chuck-Norris/

Please reply to list if it's a mailing list post - http://shlom.in/reply .

**Comment by shlomif** [comment]
Fixed now - thanks! I removed the `+ 0`.

**Comment by shlomif** [comment]
Ping! Do you need anything else?

**Review by satanas — APPROVED** [review]
Everything looks OK, Thanks! 😈

**Comment by shlomif** [comment]
On Tue, 05 Sep 2017 10:52:02 -0700
Wil Alvarez <notifications@github.com> wrote:

> Merged #4.
> 

Thanks for merging! ♥!


-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/

God signs people into the book of life using a pen that Chuck Norris gave him.
    — http://www.shlomifish.org/humour/bits/facts/Chuck-Norris/

**Pull request merged**